src/main/java/de/braintags/netrelay/controller/querypool/QueryPoolController.java

```diff
@@ -291,7 +291,7 @@ private void handleQuery(CacheEntry entry, RoutingContext context) {
    */
   private void parseDynamicQuery(DynamicQuery dynamicQuery, IQuery<?> query) throws QueryPoolException {
     if (dynamicQuery.getRootQueryPart() != null) {
-      ISearchCondition searchCondition = parseQueryParts(dynamicQuery.getRootQueryPart(), query);
+      ISearchCondition searchCondition = dynamicQuery.getRootQueryPart().toSearchCondition();
       query.setSearchCondition(searchCondition);
     }
   }
```

Comment:
Did you mean the parseDynamicQuery method? 
I still wanted to keep the parallel between parseNativeQuery and parseDynamicQuery, even if the dynamic query basically only calls the toSearchCondition() now.